tests/PHPStan/Analyser/AnalyserIntegrationTest.php

```diff
@@ -1131,4 +1131,10 @@ private function runAnalyse(string $file, ?array $allAnalysedFiles = null): arra
 		return $errors;
 	}
 
+	public function testTraitInstanceOf(): void
```

Comment:
figured a new test to make sure the newly added type-narrowing on $this in traits allows to call methods of narrowed types could be useful